### PR TITLE
net-libs/libtorrent-rasterbar: exclude broken tests + add examples

### PR DESCRIPTION
**Pull request by parona-source**
Closes: https://bugs.gentoo.org/888825
Closes: https://bugs.gentoo.org/889832
Closes: https://bugs.gentoo.org/915512

### CODE REVIEW

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Copyright 2021-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# https://github.com/arvidn/libtorrent/commit/785969574cfe925098f50b7e5d9ae2a9f2a9d0cf
```
Comment (thesamesam) [inline net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:6]: What is this saying? This is why we don't have 3.12, or...?

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		')
+	)"
+
+pkg_setup() {
+	if use python || use test; then
+		python-single-r1_pkg_setup
+	fi
+}
+
+src_configure() {
```
Comment (thesamesam) [inline net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:49]: The changes here are fine, just keep in mind that ON/OFF vs yes/no from usex _can_ make a difference if upstream use the wrong tests in `if()`.

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		pushd "${BUILD_DIR}"/examples >/dev/null || die
+		for binary in {client_test,connection_tester,custom_storage,dump_bdecode,dump_torrent,make_torrent,simple_client,stats_counters,upnp_test}; do
+			patchelf --remove-rpath ${binary} || die
+			dobin ${binary}
+		done
+		popd >/dev/null || die
+	fi
+}
+
+src_test() {
```
Comment (thesamesam) [inline net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:83]: Phase definitions should match execution order -- so put src_test before src_install.

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	use python && mycmakeargs+=( -Dboost-python-module-name="${EPYTHON/./}" )
+
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+	einstalldocs
+
+	if use examples; then
```
Comment (thesamesam) [inline net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:73]: Doesn't this really imply we shouldn't bother installing them? I guess some of them sound useful though.

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+"
+RDEPEND="${DEPEND}"
+BDEPEND="python? (
+		$(python_gen_cond_dep '
+			dev-python/setuptools[${PYTHON_USEDEP}]
+		')
+	)"
+
+pkg_setup() {
+	if use python || use test; then
```
Comment (thesamesam) [inline net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:44]: What needs Python for tests? Mention it in the commit message at least.

File: net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild (hunk 1 above)
Comment (parona-source) [reply net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:6]: > This is why we don't have 3.12

That was my intention.

Comment (thesamesam) [reply net-libs/libtorrent-rasterbar/libtorrent-rasterbar-2.0.9-r1.ebuild:6]: "no 3.12 b/c: ..."

**Pull request closed without merging**